x-pack/plugins/security_solution/public/common/components/endpoint/host_isolation/endpoint_host_isolation_status.tsx

```diff
@@ -28,12 +28,20 @@ export interface EndpointHostIsolationStatusProps {
 export const EndpointHostIsolationStatus = memo<EndpointHostIsolationStatusProps>(
   ({ isIsolated, pendingIsolate = 0, pendingUnIsolate = 0, 'data-test-subj': dataTestSubj }) => {
     const getTestId = useTestIdGenerator(dataTestSubj);
-    const isPendingStatuseDisabled = useIsExperimentalFeatureEnabled(
+    const isPendingStatusDisabled = useIsExperimentalFeatureEnabled(
       'disableIsolationUIPendingStatuses'
     );
 
+    const wasReleasing = useRef<boolean>(false);
+    const wasIsolating = useRef<boolean>(false);
```

Comment:
I'm having a hard time understanding why we need to `useRef` which given you a non-mutable const.  Looking at this logic, it seem that these two variables are set only once - when the component renders for the very first time, and then they are never updated again. If that is truly what you mean with this change, then using `useState()` like this would do the trick:

```suggestion
    const wasReleasing = useState<boolean>(pendingIsolate === 0 && pendingUnIsolate > 0);
    const wasIsolating = useState<boolean>(pendingIsolate > 0 && pendingUnIsolate === 0);
```

but I have to ask: is that really what you expect? that these will never be updated even if the component is called with different props?

Also - since the entire returned value is already memoized, you might just want to move this logic inside of the `useMemo()`'s factory callback as:

```typescript
const wasReleasing = pendingIsolate === 0 && pendingUnIsolate > 0;
const wasIsolating = pendingIsolate > 0 && pendingUnIsolate === 0;
```

______

NOTE: the reason I always pay closer attention when using `useRef` is that it can get you in trouble. Example:  see my comment below for usage in `useMemo` dependencies

Reply:
Actually, `useRef` returns a mutable object. Since, the component re-renders anyway when `pendingIsolate` and `pendingUnIsolate` change, passing the dependency array will have the same "effect". 

Overall, the idea is when the component starts to render(before returning display nodes), we want to use the old state of these values (old `isolated` badge) to decide which badge to show next. The endpoint metadata doesn't update the `Endpoint.state.isolation` value as fast as the action response indices update the pending actions count.

Reply:
Right. And since you only want it to be set on first render, you don't really need a mutable object. You can just use a `useState()` with an initial value right (like the suggestion in my prior comment)?

I'm approving since the end result I think is the same and you and confirmed that you truly only want this value to be capture on the very first render and even if they differ in subsequent re-renders, they will not be seen.